// typedb-client/stream/RequestTransmitter.js
"use strict";
/*
 * Copyright (C) 2022 Vaticle
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestTransmitter = exports.BatchDispatcher = void 0;
const RequestBuilder_1 = require("../common/rpc/RequestBuilder");
class BatchDispatcher {
    constructor(transmitter, transactionStream) {
        this._transmitter = transmitter;
        this._transactionStream = transactionStream;
        this._bufferedRequests = new Array();
        this._isRunning = false;
    }
    dispatch(req) {
        this._bufferedRequests.push(req);
        this.sendScheduledBatch();
    }
    dispatchNow(req) {
        this._bufferedRequests.push(req);
        this.sendNow();
    }
    close() {
        this._transmitter._dispatchers.delete(this);
        this._transactionStream.end();
    }
    sendNow() {
        const clientRequest = RequestBuilder_1.RequestBuilder.Transaction.clientReq(this._bufferedRequests);
        this._transactionStream.write(clientRequest);
        this._bufferedRequests = [];
    }
    sendScheduledBatch() {
        if (this._isRunning)
            return;
        this._isRunning = true;
        this.setSchedule(true);
    }
    setSchedule(first) {
        const wait = first ? BatchDispatcher.BATCH_WINDOW_SMALL_MILLIS : BatchDispatcher.BATCH_WINDOW_LARGE_MILLIS;
        setTimeout(() => {
            if (this._bufferedRequests.length > 0) {
                this.sendNow();
                this.setSchedule(false);
            }
            else {
                this._isRunning = false;
            }
        }, wait);
    }
}
exports.BatchDispatcher = BatchDispatcher;
BatchDispatcher.BATCH_WINDOW_SMALL_MILLIS = 1;
BatchDispatcher.BATCH_WINDOW_LARGE_MILLIS = 3;
class RequestTransmitter {
    constructor() {
        this._dispatchers = new Set();
        this._isOpen = true;
    }
    close() {
        if (this._isOpen) {
            this._isOpen = false;
            this._dispatchers.forEach(dispatcher => dispatcher.close());
        }
    }
    dispatcher(transactionStream) {
        const dispatcher = new BatchDispatcher(this, transactionStream);
        this._dispatchers.add(dispatcher);
        return dispatcher;
    }
}
exports.RequestTransmitter = RequestTransmitter;
