// typedb-protocol/common/options_pb.js
// source: common/options.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

goog.exportSymbol('proto.typedb.protocol.Options', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.ExplainOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.InferOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.ParallelOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.PrefetchOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.PrefetchSizeOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.ReadAnyReplicaOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.SchemaLockAcquireTimeoutOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.SessionIdleTimeoutOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.TraceInferenceOptCase', null, global);
goog.exportSymbol('proto.typedb.protocol.Options.TransactionTimeoutOptCase', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.typedb.protocol.Options = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.typedb.protocol.Options.oneofGroups_);
};
goog.inherits(proto.typedb.protocol.Options, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.typedb.protocol.Options.displayName = 'proto.typedb.protocol.Options';
}

/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.typedb.protocol.Options.oneofGroups_ = [[1],[2],[3],[4],[5],[6],[7],[8],[9],[10]];

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.InferOptCase = {
  INFER_OPT_NOT_SET: 0,
  INFER: 1
};

/**
 * @return {proto.typedb.protocol.Options.InferOptCase}
 */
proto.typedb.protocol.Options.prototype.getInferOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.InferOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[0]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.TraceInferenceOptCase = {
  TRACE_INFERENCE_OPT_NOT_SET: 0,
  TRACE_INFERENCE: 2
};

/**
 * @return {proto.typedb.protocol.Options.TraceInferenceOptCase}
 */
proto.typedb.protocol.Options.prototype.getTraceInferenceOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.TraceInferenceOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[1]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.ExplainOptCase = {
  EXPLAIN_OPT_NOT_SET: 0,
  EXPLAIN: 3
};

/**
 * @return {proto.typedb.protocol.Options.ExplainOptCase}
 */
proto.typedb.protocol.Options.prototype.getExplainOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.ExplainOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[2]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.ParallelOptCase = {
  PARALLEL_OPT_NOT_SET: 0,
  PARALLEL: 4
};

/**
 * @return {proto.typedb.protocol.Options.ParallelOptCase}
 */
proto.typedb.protocol.Options.prototype.getParallelOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.ParallelOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[3]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.PrefetchSizeOptCase = {
  PREFETCH_SIZE_OPT_NOT_SET: 0,
  PREFETCH_SIZE: 5
};

/**
 * @return {proto.typedb.protocol.Options.PrefetchSizeOptCase}
 */
proto.typedb.protocol.Options.prototype.getPrefetchSizeOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.PrefetchSizeOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[4]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.PrefetchOptCase = {
  PREFETCH_OPT_NOT_SET: 0,
  PREFETCH: 6
};

/**
 * @return {proto.typedb.protocol.Options.PrefetchOptCase}
 */
proto.typedb.protocol.Options.prototype.getPrefetchOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.PrefetchOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[5]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.SessionIdleTimeoutOptCase = {
  SESSION_IDLE_TIMEOUT_OPT_NOT_SET: 0,
  SESSION_IDLE_TIMEOUT_MILLIS: 7
};

/**
 * @return {proto.typedb.protocol.Options.SessionIdleTimeoutOptCase}
 */
proto.typedb.protocol.Options.prototype.getSessionIdleTimeoutOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.SessionIdleTimeoutOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[6]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.TransactionTimeoutOptCase = {
  TRANSACTION_TIMEOUT_OPT_NOT_SET: 0,
  TRANSACTION_TIMEOUT_MILLIS: 8
};

/**
 * @return {proto.typedb.protocol.Options.TransactionTimeoutOptCase}
 */
proto.typedb.protocol.Options.prototype.getTransactionTimeoutOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.TransactionTimeoutOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[7]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.SchemaLockAcquireTimeoutOptCase = {
  SCHEMA_LOCK_ACQUIRE_TIMEOUT_OPT_NOT_SET: 0,
  SCHEMA_LOCK_ACQUIRE_TIMEOUT_MILLIS: 9
};

/**
 * @return {proto.typedb.protocol.Options.SchemaLockAcquireTimeoutOptCase}
 */
proto.typedb.protocol.Options.prototype.getSchemaLockAcquireTimeoutOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.SchemaLockAcquireTimeoutOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[8]));
};

/**
 * @enum {number}
 */
proto.typedb.protocol.Options.ReadAnyReplicaOptCase = {
  READ_ANY_REPLICA_OPT_NOT_SET: 0,
  READ_ANY_REPLICA: 10
};

/**
 * @return {proto.typedb.protocol.Options.ReadAnyReplicaOptCase}
 */
proto.typedb.protocol.Options.prototype.getReadAnyReplicaOptCase = function() {
  return /** @type {proto.typedb.protocol.Options.ReadAnyReplicaOptCase} */(jspb.Message.computeOneofCase(this, proto.typedb.protocol.Options.oneofGroups_[9]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.typedb.protocol.Options.prototype.toObject = function(opt_includeInstance) {
  return proto.typedb.protocol.Options.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.typedb.protocol.Options} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.typedb.protocol.Options.toObject = function(includeInstance, msg) {
  var f, obj = {
    infer: jspb.Message.getBooleanFieldWithDefault(msg, 1, false),
    traceInference: jspb.Message.getBooleanFieldWithDefault(msg, 2, false),
    explain: jspb.Message.getBooleanFieldWithDefault(msg, 3, false),
    parallel: jspb.Message.getBooleanFieldWithDefault(msg, 4, false),
    prefetchSize: jspb.Message.getFieldWithDefault(msg, 5, 0),
    prefetch: jspb.Message.getBooleanFieldWithDefault(msg, 6, false),
    sessionIdleTimeoutMillis: jspb.Message.getFieldWithDefault(msg, 7, 0),
    transactionTimeoutMillis: jspb.Message.getFieldWithDefault(msg, 8, 0),
    schemaLockAcquireTimeoutMillis: jspb.Message.getFieldWithDefault(msg, 9, 0),
    readAnyReplica: jspb.Message.getBooleanFieldWithDefault(msg, 10, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.typedb.protocol.Options}
 */
proto.typedb.protocol.Options.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.typedb.protocol.Options;
  return proto.typedb.protocol.Options.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.typedb.protocol.Options} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.typedb.protocol.Options}
 */
proto.typedb.protocol.Options.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setInfer(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setTraceInference(value);
      break;
    case 3:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setExplain(value);
      break;
    case 4:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setParallel(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setPrefetchSize(value);
      break;
    case 6:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setPrefetch(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setSessionIdleTimeoutMillis(value);
      break;
    case 8:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setTransactionTimeoutMillis(value);
      break;
    case 9:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setSchemaLockAcquireTimeoutMillis(value);
      break;
    case 10:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setReadAnyReplica(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.typedb.protocol.Options.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.typedb.protocol.Options.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.typedb.protocol.Options} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.typedb.protocol.Options.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {boolean} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeBool(
      1,
      f
    );
  }
  f = /** @type {boolean} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeBool(
      2,
      f
    );
  }
  f = /** @type {boolean} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeBool(
      3,
      f
    );
  }
  f = /** @type {boolean} */ (jspb.Message.getField(message, 4));
  if (f != null) {
    writer.writeBool(
      4,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 5));
  if (f != null) {
    writer.writeInt32(
      5,
      f
    );
  }
  f = /** @type {boolean} */ (jspb.Message.getField(message, 6));
  if (f != null) {
    writer.writeBool(
      6,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 7));
  if (f != null) {
    writer.writeInt32(
      7,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 8));
  if (f != null) {
    writer.writeInt32(
      8,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 9));
  if (f != null) {
    writer.writeInt32(
      9,
      f
    );
  }
  f = /** @type {boolean} */ (jspb.Message.getField(message, 10));
  if (f != null) {
    writer.writeBool(
      10,
      f
    );
  }
};


/**
 * optional bool infer = 1;
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.getInfer = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setInfer = function(value) {
  return jspb.Message.setOneofField(this, 1, proto.typedb.protocol.Options.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearInfer = function() {
  return jspb.Message.setOneofField(this, 1, proto.typedb.protocol.Options.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasInfer = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional bool trace_inference = 2;
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.getTraceInference = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setTraceInference = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.typedb.protocol.Options.oneofGroups_[1], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearTraceInference = function() {
  return jspb.Message.setOneofField(this, 2, proto.typedb.protocol.Options.oneofGroups_[1], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasTraceInference = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional bool explain = 3;
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.getExplain = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 3, false));
};


/**
 * @param {boolean} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setExplain = function(value) {
  return jspb.Message.setOneofField(this, 3, proto.typedb.protocol.Options.oneofGroups_[2], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearExplain = function() {
  return jspb.Message.setOneofField(this, 3, proto.typedb.protocol.Options.oneofGroups_[2], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasExplain = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional bool parallel = 4;
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.getParallel = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 4, false));
};


/**
 * @param {boolean} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setParallel = function(value) {
  return jspb.Message.setOneofField(this, 4, proto.typedb.protocol.Options.oneofGroups_[3], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearParallel = function() {
  return jspb.Message.setOneofField(this, 4, proto.typedb.protocol.Options.oneofGroups_[3], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasParallel = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional int32 prefetch_size = 5;
 * @return {number}
 */
proto.typedb.protocol.Options.prototype.getPrefetchSize = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setPrefetchSize = function(value) {
  return jspb.Message.setOneofField(this, 5, proto.typedb.protocol.Options.oneofGroups_[4], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearPrefetchSize = function() {
  return jspb.Message.setOneofField(this, 5, proto.typedb.protocol.Options.oneofGroups_[4], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasPrefetchSize = function() {
  return jspb.Message.getField(this, 5) != null;
};


/**
 * optional bool prefetch = 6;
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.getPrefetch = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 6, false));
};


/**
 * @param {boolean} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setPrefetch = function(value) {
  return jspb.Message.setOneofField(this, 6, proto.typedb.protocol.Options.oneofGroups_[5], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearPrefetch = function() {
  return jspb.Message.setOneofField(this, 6, proto.typedb.protocol.Options.oneofGroups_[5], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasPrefetch = function() {
  return jspb.Message.getField(this, 6) != null;
};


/**
 * optional int32 session_idle_timeout_millis = 7;
 * @return {number}
 */
proto.typedb.protocol.Options.prototype.getSessionIdleTimeoutMillis = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setSessionIdleTimeoutMillis = function(value) {
  return jspb.Message.setOneofField(this, 7, proto.typedb.protocol.Options.oneofGroups_[6], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearSessionIdleTimeoutMillis = function() {
  return jspb.Message.setOneofField(this, 7, proto.typedb.protocol.Options.oneofGroups_[6], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasSessionIdleTimeoutMillis = function() {
  return jspb.Message.getField(this, 7) != null;
};


/**
 * optional int32 transaction_timeout_millis = 8;
 * @return {number}
 */
proto.typedb.protocol.Options.prototype.getTransactionTimeoutMillis = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 8, 0));
};


/**
 * @param {number} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setTransactionTimeoutMillis = function(value) {
  return jspb.Message.setOneofField(this, 8, proto.typedb.protocol.Options.oneofGroups_[7], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearTransactionTimeoutMillis = function() {
  return jspb.Message.setOneofField(this, 8, proto.typedb.protocol.Options.oneofGroups_[7], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasTransactionTimeoutMillis = function() {
  return jspb.Message.getField(this, 8) != null;
};


/**
 * optional int32 schema_lock_acquire_timeout_millis = 9;
 * @return {number}
 */
proto.typedb.protocol.Options.prototype.getSchemaLockAcquireTimeoutMillis = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 9, 0));
};


/**
 * @param {number} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setSchemaLockAcquireTimeoutMillis = function(value) {
  return jspb.Message.setOneofField(this, 9, proto.typedb.protocol.Options.oneofGroups_[8], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearSchemaLockAcquireTimeoutMillis = function() {
  return jspb.Message.setOneofField(this, 9, proto.typedb.protocol.Options.oneofGroups_[8], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasSchemaLockAcquireTimeoutMillis = function() {
  return jspb.Message.getField(this, 9) != null;
};


/**
 * optional bool read_any_replica = 10;
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.getReadAnyReplica = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 10, false));
};


/**
 * @param {boolean} value
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.setReadAnyReplica = function(value) {
  return jspb.Message.setOneofField(this, 10, proto.typedb.protocol.Options.oneofGroups_[9], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.typedb.protocol.Options} returns this
 */
proto.typedb.protocol.Options.prototype.clearReadAnyReplica = function() {
  return jspb.Message.setOneofField(this, 10, proto.typedb.protocol.Options.oneofGroups_[9], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.typedb.protocol.Options.prototype.hasReadAnyReplica = function() {
  return jspb.Message.getField(this, 10) != null;
};


goog.object.extend(exports, proto.typedb.protocol);
