// typedb-client/common/Label.js
"use strict";
/*
 * Copyright (C) 2022 Vaticle
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Label = void 0;
class Label {
    constructor(scope, label) {
        this._scope = scope;
        this._name = label;
    }
    get scope() {
        return this._scope;
    }
    get name() {
        return this._name;
    }
    get scopedName() {
        return this._scope == null ? this._name : `${this._scope}:${this._name}`;
    }
    toString() {
        return this.scopedName;
    }
    equals(that) {
        return this._scope === that._scope && this._name === that._name;
    }
}
exports.Label = Label;
(function (Label) {
    function of(name) {
        return new Label(null, name);
    }
    Label.of = of;
    function scoped(scope, name) {
        return new Label(scope, name);
    }
    Label.scoped = scoped;
})(Label = exports.Label || (exports.Label = {}));
