// typedb-client/connection/cluster/ClusterUser.js
"use strict";
/*
 * Copyright (C) 2022 Vaticle
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterUser = void 0;
const RequestBuilder_1 = require("../../common/rpc/RequestBuilder");
const dependencies_internal_1 = require("../../dependencies_internal");
class ClusterUser {
    constructor(client, username) {
        this._client = client;
        this._username = username;
    }
    async password(password) {
        const failsafeTask = new ClusterUserFailsafeTask(this._client, (replica) => {
            return this._client.stub(replica.address).userPassword(RequestBuilder_1.RequestBuilder.Cluster.User.passwordReq(this.username, password));
        });
        await failsafeTask.runPrimaryReplica();
    }
    async delete() {
        const failsafeTask = new ClusterUserFailsafeTask(this._client, (replica) => {
            return this._client.stub(replica.address).userDelete(RequestBuilder_1.RequestBuilder.Cluster.User.deleteReq(this.username));
        });
        await failsafeTask.runPrimaryReplica();
    }
    get username() {
        return this._username;
    }
}
exports.ClusterUser = ClusterUser;
class ClusterUserFailsafeTask extends dependencies_internal_1.FailsafeTask {
    constructor(client, task) {
        super(client, dependencies_internal_1.ClusterUserManager._SYSTEM_DB);
        this._task = task;
    }
    run(replica) {
        return this._task(replica);
    }
}
