// typedb-client/connection/cluster/ClusterUserManager.js
"use strict";
/*
 * Copyright (C) 2022 Vaticle
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClusterUserManager = void 0;
const ErrorMessage_1 = require("../../common/errors/ErrorMessage");
const TypeDBClientError_1 = require("../../common/errors/TypeDBClientError");
const RequestBuilder_1 = require("../../common/rpc/RequestBuilder");
const dependencies_internal_1 = require("../../dependencies_internal");
var CLUSTER_USER_DOES_NOT_EXIST = ErrorMessage_1.ErrorMessage.Client.CLUSTER_USER_DOES_NOT_EXIST;
class ClusterUserManager {
    constructor(client) {
        this._client = client;
    }
    all() {
        const failsafeTask = new UserManagerFailsafeTask(this._client, (replica) => {
            return this._client.stub(replica.address).usersAll(RequestBuilder_1.RequestBuilder.Cluster.UserManager.allReq())
                .then((res) => {
                return res.getNamesList().map(name => new dependencies_internal_1.ClusterUser(this._client, name));
            });
        });
        return failsafeTask.runPrimaryReplica();
    }
    contains(username) {
        const failsafeTask = new UserManagerFailsafeTask(this._client, (replica) => {
            return this._client.stub(replica.address).usersContains(RequestBuilder_1.RequestBuilder.Cluster.UserManager.containsReq(username));
        });
        return failsafeTask.runPrimaryReplica();
    }
    create(username, password) {
        const failsafeTask = new UserManagerFailsafeTask(this._client, (replica) => {
            return this._client.stub(replica.address).userCreate(RequestBuilder_1.RequestBuilder.Cluster.UserManager.createReq(username, password));
        });
        return failsafeTask.runPrimaryReplica();
    }
    async get(username) {
        if (await this.contains(username))
            return new dependencies_internal_1.ClusterUser(this._client, username);
        else
            throw new TypeDBClientError_1.TypeDBClientError(CLUSTER_USER_DOES_NOT_EXIST.message(username));
    }
}
exports.ClusterUserManager = ClusterUserManager;
ClusterUserManager._SYSTEM_DB = "_system";
class UserManagerFailsafeTask extends dependencies_internal_1.FailsafeTask {
    constructor(client, task) {
        super(client, ClusterUserManager._SYSTEM_DB);
        this._task = task;
    }
    run(replica) {
        return this._task(replica);
    }
}
